r4e-image.sh

```diff
@@ -1,18 +1,27 @@
+#!/bin/bash
 # script for creating an image that points to a certain ostree commit from web server
 # the image includes the agent RPM that communicates with server that monitors edge devices
 # at the end the following products will be available for download:
 #   - ostree image
 #   - kickstart file
 # input:
-#  - IP or hostname with port if different than 80 used for connecting to the local web server
-#  - a name for the image. allows using different ostree image for different types of edge devices
-#  - URL for connecting to the server that monitors edge devices
-#  - URL for additional packages repository
+#  - <image-server-address> - IP or hostname with port if different than 80 used for connecting to the local web server
+#  - <image-name> - a name for the image. allows using different ostree image for different types of edge devices
+#  - <agent-server-url> - URL for connecting to the server that monitors edge devices
+#  - [packages-repo-url] - optional, URL for additional packages repository
 # script assumes that files `blueprint.template` and `ks.cfg.template` are present in the working directory
+
+function cleanup() {
+    rm -f temp.out 2 &> /dev/null
+    rm -f $BUILD_ID-commit.tar 2 &> /dev/null
+    rm -f $REPO_DIR 2 &> /dev/null
+}
+trap 'cleanup' EXIT
+
 set -e
 Usage() {
     echo "Usage:"
-    echo `basename $0` "<image-server-address> <image-name> <agent-server-url> [packages-repo-url]"
+    echo $(basename "$0") "<image-server-address> <image-name> <agent-server-url> [packages-repo-url]"
```

Comment:
out of curiosity, what is the reason for changing from backticks (``) to '$()'?

Reply:
`$()` is POSIX, while backticks are legacy. In general there are few advantages other than that:
you can have nesting commands with `$(ls $(echo $dir))` and evaluation on execution and not no declaration.

Reply:
Legacy? It has been in use for decades and will remain for a few decades more